doc/kernel.rst.kerneldoc

```diff
@@ -0,0 +1,33 @@
+..
+    Zephyr Project Kernel Documentation master file
+
+Zephyr Project Kernel Documentation
+###################################
+
+This document provides a general introduction of the Zephyr kernel's
+key capabilities and services.
+
+.. toctree::
+   :maxdepth: 3
+
+   kernel/kernel.rst
+
+.. toctree::
+   :maxdepth: 2
+
+   api/api.rst
+
+.. toctree::
+   :maxdepth: 1
+
+   reference/kconfig/index.rst
+
+Links
+^^^^^
```

Comment:
This isn't following the Zephyr documentation convention for heading hierarchy:  ``# * = -``